tests/test_parallel_graceful.py

```diff
@@ -0,0 +1,223 @@
+from typing import List
+
+from hamilton import ad_hoc_utils, driver
+from hamilton.execution.executors import SynchronousLocalTaskExecutor
+from hamilton.function_modifiers import config, tag
+from hamilton.htypes import Collect, Parallelizable
+from hamilton.lifecycle import GracefulErrorAdapter
+
+
+@config.when(test_front="good")
+def input_maker__good(n: int) -> int:
+    return n
+
+
+@config.when(test_front="bad")
+def input_maker__bad(n: int) -> int:
+    raise Exception("Went wrong")
+
+
+@config.when(test_state="middle")
+def distro__middle(input_maker: int) -> Parallelizable[int]:
+    for x in range(input_maker):
+        if x > 4:
+            raise Exception("bad")
+        yield x * 3
+
+
+@config.when(test_state="early")
+def distro__early(input_maker: int) -> Parallelizable[int]:
+    raise Exception("bad")
+    for x in range(input_maker):
+        yield x * 3
+
+
+@config.when(test_state="pass")
+def distro__pass(input_maker: int) -> Parallelizable[int]:
+    for x in range(input_maker):
+        yield x * 3
+
+
+def some_math(distro: int) -> float:
+    if distro > 15:
+        raise Exception("No no no")
+    return distro * 2.0
+
+
+def other_math(distro: int) -> float:
+    if distro < 10:
+        raise Exception("Not allowed")
+    return distro + 1
+
+
+@tag(allow_injection="allow")
+def gather_math(some_math: float, other_math: float) -> List[float]:
+    return [some_math, other_math]
+
+
+def distro_end(gather_math: Collect[List[float]]) -> List[float]:
+    ans = [x for x in gather_math]
+    return ans
+
+
+def distro_gather(some_math: Collect[float]) -> List[float]:
+    ans = [x for x in some_math]
+    return ans
+
+
+# TODO: How does this break extract_fields, e.g.?
+
+temp_module_simple = ad_hoc_utils.create_temporary_module(
+    input_maker__good,
+    input_maker__bad,
+    distro__early,
+    distro__middle,
+    distro__pass,
+    some_math,
+    distro_gather,
+    module_name="parallel_example",
+)
+
+temp_module = ad_hoc_utils.create_temporary_module(
```

Comment:
So I actually like `create_temporary_module` better for this, but it might be worth putting these two modules in `resources` (as it's consistent with the rest of the way we're doing things): https://github.com/DAGWorks-Inc/hamilton/tree/main/tests/resources.

Reply:
Moved the module over to resources.

[a299289](https://github.com/DAGWorks-Inc/hamilton/pull/1017/commits/a299289c71558537909606e6c817a75280efa8f6)